stablehlo/tests/interpret_slice.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: stablehlo-interpreter --interpret -split-input-file %s | FileCheck %s
```

Comment:
Same question as in https://github.com/openxla/stablehlo/pull/1006/files#r1090063327. How do we decide how many of these tests are needed? E.g. should we only have `slice_op_2d` since it kind of tests everything that other tests include? 

And furthermore, how do we write positive tests in general? Also paging @sdasgup3 to discuss this.

Reply:
Good question. Should we include one comprehensive test that covers all the nuances or test the behaviors separately. E.g. DynamicUpdateSlice, testing that the op with the simplest formulation works, testing clamping of start indices, etc. or include them bunch in a single test.

Reply:
>> how many test
I would say that it desirable to have minimal tests which meets our set coverage criteria. For the 

>> how to decide   
The idea behind the testing is to make sure that the spec is comprehensively covered. That coverage must also get transcended to the interpreter implementation as it is supposed to be readable implementation of the spec. having said that the number we should aim for close to minimum while meeting our coverage criteria.

For a like SliceOp,  we could have a single test covering all the cases. This will not be the case if for an op if its implementation handles dimensions differently: That needs separate test to cover each code path. 
Taking assistance from the coverage tools might help us when to stop. 

 
I propose to write test with "best effort" covering  multiple behaviors. By doing that I believe most of the simpler ops will end up having one test. From collective "best efforts" from reviewer and author, I think we should be able to get a near minimal set

Reply:
LGTM. Let's update the checklist accordingly?

Reply:
Given the discussion above, do we still want to have the `slice_op_empty_slice` test?

Reply:
Agreed. Will remove the test.